pkg/shares/share_merging_test.go

```diff
@@ -0,0 +1,146 @@
+package shares
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/nmt/namespace"
+)
+
+func TestParseShares(t *testing.T) {
+	type testCase struct {
+		name      string
+		shares    [][]byte
+		want      []ShareSequence
+		expectErr bool
+	}
+
+	start := true
+	continuation := false
+	messageOneNamespace := namespace.ID{1, 1, 1, 1, 1, 1, 1, 1}
+	messageTwoNamespace := namespace.ID{2, 2, 2, 2, 2, 2, 2, 2}
+
+	transactionShareStart := generateRawShare(appconsts.TxNamespaceID, start)
+	transactionShareContinuation := generateRawShare(appconsts.TxNamespaceID, continuation)
+
+	evidenceShareStart := generateRawShare(appconsts.EvidenceNamespaceID, start)
+	evidenceShareContinuation := generateRawShare(appconsts.EvidenceNamespaceID, continuation)
+
+	messageOneStart := generateRawShare(messageOneNamespace, start)
+	messageOneContinuation := generateRawShare(messageOneNamespace, continuation)
+
+	messageTwoStart := generateRawShare(messageTwoNamespace, start)
+	messageTwoContinuation := generateRawShare(messageTwoNamespace, continuation)
+
+	invalidShare := generateRawShare(messageOneNamespace, start)
+	invalidShare = append(invalidShare, []byte{0}...)
```

Comment:
[blocking]
perhaps we could modify this test to use the share splitting api so that we know if we break this in the future?

Reply:
Good idea. Resolved in https://github.com/celestiaorg/celestia-app/pull/828/commits/c184932f0926d119de14ce30bc2611d75f2b1b3e